src/shogun/lib/SGVector.cpp

```diff
@@ -114,40 +114,48 @@ SGVector<T>::~SGVector()
 
 template <class T>
 SGVector<T>::SGVector(EigenVectorXt& vec)
-: SGReferencedData(false), vector(vec.data()), vlen(vec.size())
+: SGReferencedData(false), vector(vec.data()), vlen(vec.size()), gpu_vector(NULL)
 {
 
 }
 
 template <class T>
 SGVector<T>::SGVector(EigenRowVectorXt& vec)
-: SGReferencedData(false), vector(vec.data()), vlen(vec.size())
+: SGReferencedData(false), vector(vec.data()), vlen(vec.size()), gpu_vector(NULL)
 {
 
 }
 
 template <class T>
 SGVector<T>::operator EigenVectorXtMap() const
 {
+	if (on_gpu())
+		SG_SERROR("EigenRowVectorXtMap method not available for vectors on GPU.\n")
 	return EigenVectorXtMap(vector, vlen);
 }
 
 template <class T>
 SGVector<T>::operator EigenRowVectorXtMap() const
 {
+	if (on_gpu())
+		SG_SERROR("EigenRowVectorXtMap method not available for vectors on GPU.\n")
 	return EigenRowVectorXtMap(vector, vlen);
 }
 
 template<class T>
 void SGVector<T>::zero()
 {
+	if (on_gpu())
+		SG_SERROR("Zero method not available for vectors on GPU.\n")
```

Comment:
or possibly just for being consistent it might be better to remove the ::zero() method and have it in the `linalg::<backend>::zero(SGVector<T>& v);`